src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java

```diff
@@ -18,27 +19,62 @@
   /**
-   * Creates a new ExampleSubsystem.
+   * Creates a new DriveBaseSubsystem.
    */
 
   private final Joystick mDriverJoystick;
   private final CANSparkMax mRightFront;
+  private final CANSparkMax mRightMiddle;
   private final CANSparkMax mRightRear;
   private final CANSparkMax mLeftFront;
+  private final CANSparkMax mLeftMiddle;
   private final CANSparkMax mLeftRear;
   private final DifferentialDrive mDiffDrive;
+  private final CANEncoder mRightFrontEncoder;
+  private final CANEncoder mRightMiddleEncoder;
+  private final CANEncoder mRightRearEncoder;
+  private final CANEncoder mLeftFrontEncoder;
+  private final CANEncoder mLeftMiddleEncoder;
+  private final CANEncoder mLeftRearEncoder;
   private boolean mIsArcadeDrive = true;
 
-  public DriveBaseSubsystem(Joystick js) {
-    mDriverJoystick = js;
+  public DriveBaseSubsystem(Joystick joystick) {
+    mDriverJoystick = joystick;
     mRightFront = new CANSparkMaxWrap(Constants.kDriveRightFront, MotorType.kBrushless);
+    mRightMiddle = new CANSparkMaxWrap(Constants.kDriveRightMiddle, MotorType.kBrushless);
     mRightRear = new CANSparkMaxWrap(Constants.kDriveRightRear, MotorType.kBrushless);
     mLeftFront = new CANSparkMaxWrap(Constants.kDriveLeftFront, MotorType.kBrushless);
+    mLeftMiddle = new CANSparkMaxWrap(Constants.kDriveLeftMiddle, MotorType.kBrushless);
     mLeftRear = new CANSparkMaxWrap(Constants.kDriveLeftRear, MotorType.kBrushless);
+    //The gear boxes are mirrored and one side needs to be inverted
     mLeftFront.setInverted(true);
+    mLeftMiddle.setInverted(true);
     mLeftRear.setInverted(true);
     mLeftRear.follow(mLeftFront);
+    mLeftMiddle.follow(mLeftFront);
     mRightRear.follow(mRightFront);
+    mRightMiddle.follow(mRightFront);
+    //Reset the flash memory on the spark maxes so any saved configuration values do not carry over
+    //We are configuring the spark maxes below
+    mRightFront.restoreFactoryDefaults();
+    mRightMiddle.restoreFactoryDefaults();
+    mRightRear.restoreFactoryDefaults();
+    mLeftFront.restoreFactoryDefaults();
+    mLeftMiddle.restoreFactoryDefaults();
+    mLeftRear.restoreFactoryDefaults();
+    //Current limiting is required to prevent brown outs
+    mRightFront.setSecondaryCurrentLimit(Constants.kDriveBaseCurrentLimit);
+    mRightMiddle.setSecondaryCurrentLimit(Constants.kDriveBaseCurrentLimit);
+    mRightRear.setSecondaryCurrentLimit(Constants.kDriveBaseCurrentLimit);
+    mLeftFront.setSecondaryCurrentLimit(Constants.kDriveBaseCurrentLimit);
+    mLeftMiddle.setSecondaryCurrentLimit(Constants.kDriveBaseCurrentLimit);
+    mLeftRear.setSecondaryCurrentLimit(Constants.kDriveBaseCurrentLimit);
+    mRightFrontEncoder = mRightFront.getEncoder();
+    mRightMiddleEncoder = mRightMiddle.getEncoder();
+    mRightRearEncoder = mRightRear.getEncoder();
+    mLeftFrontEncoder = mLeftFront.getEncoder();
+    mLeftMiddleEncoder = mLeftMiddle.getEncoder();
+    mLeftRearEncoder = mLeftRear.getEncoder();
```

Comment:
This sort of copy and paste code is where making an array of CANSparkMax objects and then using loops would be helpful.

Reply:
Created a new issue #9 to track, because I don't think this should block the PR and should be a future upgrade instead.